datastore/gcloud/aio/datastore/query.py

```diff
@@ -5,11 +5,60 @@
 from gcloud.aio.datastore.constants import MoreResultsType
 from gcloud.aio.datastore.constants import ResultType
 from gcloud.aio.datastore.entity import EntityResult
+from gcloud.aio.datastore.filter import Filter
 from gcloud.aio.datastore.utils import make_value
 from gcloud.aio.datastore.utils import parse_value
 
 
-class GQLQuery:
+class _BaseQuery:
+    json_key = ''
+
+    def __repr__(self) -> str:
+        return str(self.to_repr())
+
+    @classmethod
+    def from_repr(cls, data: Dict[str, Any]) -> '_BaseQuery':
+        raise NotImplementedError
+
+    def to_repr(self) -> Dict[str, Any]:
+        raise NotImplementedError
+
+
+# https://cloud.google.com/datastore/docs/reference/data/rest/v1/projects/runQuery#Query
+class Query(_BaseQuery):
+    json_key = 'query'
+
+    def __init__(self, kind: str, query_filter: Filter = None):
+        if not kind:
+            raise NotImplementedError('Kindless queries are not supported')
```

Comment:
It'd be nice to implement this. If I'm reading the docs correctly, looks like we'd just need to change the `to_repr` method to attach an empty list. Something like:
```
data = {'kind': [{'name': self.kind}] if self.kind else []}
```